src/signals.c

```diff
@@ -39,3 +40,21 @@ void	sigquit_handler_in_child(int sig)
 		ft_printf("Quit\n");
 	}
 }
+
+void	sigint_handler_after_here_doc(int sig)
+{
+	if (sig == SIGINT)
+    {
+		rl_on_new_line();
+		rl_replace_line("", 0);
+		rl_redisplay();
+		g_exit = sig;
+	}
+}
+
+void	sigint_handler_here_doc(int sig)
+{
+	if (sig == SIGINT)
+	g_exit = SIGINT;
+	close(STDIN_FILENO);
```

Comment:
This is clever, I couldn't stop the deadline sending EOF or anything

Reply:
I tried so many different ways to make readline stops... and I knew it needs EOF and the only way I figured out to "fake" a EOF is by closing the STDIN. Problem is that you need to open the STDIN as soon as you're done with the exit procedure caused by SIGINT, hence the open("/dev/tty", O_RDONLY)... which is basically a way to undo the close(STDIN_FILENO). Thing is that opening /dev/tty ignores any other fd redirection that is in place... but this only happens with here_doc, so it should be ok

Reply:
I tried using `ioctl(STDIN_FILENO, TIOCSTI, "\004");` that sends EOF to the terminal, but it did not close deadline. I will ask in school